OSA/github_agent/github_agent.py

```diff
@@ -0,0 +1,133 @@
+from git import Repo, GitCommandError, InvalidGitRepositoryError
+import os
+import logging
+import requests
+from dotenv import load_dotenv
+for handler in logging.root.handlers[:]:
+    logging.root.removeHandler(handler)
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(message)s",
+)
+
+
+class GithubAgent:
+    def __init__(self, repo_url: str, branch_name: str = "OSA"):
+        """
+        Initializes GithubAgent.
+
+        :param repo_url: URL of the repository.
+        :param branch_name: Name of the branch to be created.
+        """
+        load_dotenv()
+        self.repo_url = repo_url
+        self.clone_dir = repo_url.rstrip('/').split('/')[-1]
+        self.branch_name = branch_name
+        self.repo = None
+        self.token = os.getenv("GIT_TOKEN")
+
+    def clone_repository(self) -> None:
+        """
+        Clones the repository into the specified directory if it doesn't exist locally.
+        If the repository already exists locally, initializes it.
+
+        :raises InvalidGitRepositoryError: If the local directory exists but is not a valid Git repository.
+        :raises GitCommandError: If cloning the repository fails.
+        """
+        if self.repo:
+            logging.warning("Repository already initialized")
```

Comment:
That would be more informative

```suggestion
            logging.warning(f"A repository already initialized ({self.repo_url})")
```

Reply:
Поправил